pif_ingestor/core.py

```diff
@@ -1,42 +1,65 @@
 from .ui import get_cli
-from .manager import run_extension
+from .manager import IngesterManager
 from .enrichment import add_tags, add_license, add_contact
 from .uploader import upload
 from .packager import create_package
 import os.path
+from os import walk
 from pypif import pif
 import logging
 
 
-def main():
-    """Main driver for pif-ingestor"""
-
-    # Get direction from the user
-    parser = get_cli()
-    args = parser.parse_args()
-
+def _handle_pif(path, ingest_name, convert_args, enrich_args, ingest_manager):
+    """Ingest and enrich pifs from a path, returning affected paths"""
     # Run an ingest extension
-    pifs = run_extension(args.format, args.path, args.converter_arguments)
+    pifs = ingest_manager.run_extension(ingest_name, path, convert_args)
```

Comment:
They don't need to be bubbled.  This is a user-invoked script, so an error in the extension will be raised to the user and presented with a stack trace.